src/Framework/AspNetCoreAnalyzers/src/Analyzers/RouteHandlers/DetectMismatchedParameterOptionality.cs

```diff
@@ -21,7 +22,16 @@ public partial class RouteHandlerAnalyzer : DiagnosticAnalyzer
             return;
         }
 
-        var value = invocation.Arguments[1].Value;
+        IOperation? value = null;
+        foreach (var argument in invocation.Arguments)
+        {
+            if (argument.Parameter.Ordinal == 1)
+            {
+                value = invocation.Arguments[1].Value;
```

Comment:
This looks strange, should it be `argument.Value`?

Reply:
Nice catch! Indeed this is a copy/paste error.